ogc-example/src/main/scala/geotrellis/server/ogc/FocalParameters.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2020 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package geotrellis.server.ogc
+
+import geotrellis.server.ogc.params.{ParamError, ParamMap}
+import geotrellis.server.ogc.utils.ExpressionUtils
+
+import geotrellis.raster.TargetCell
+import com.azavea.maml.ast.{Expression, FocalHillshade, FocalSlope}
+import cats.Semigroup
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.Valid
+import cats.syntax.option._
+import cats.syntax.apply._
+import cats.instances.option._
+import cats.syntax.semigroup._
+
+case class FocalParameters(azimuth: Option[Double], altitude: Option[Double], zFactor: Option[Double], target: Option[TargetCell]) {
+  import FocalParameters._
+
+  def bind: Expression => Expression = {
+    case e @ FocalHillshade(_, az, al, z, t) => e.copy(
+      azimuth  = azimuth.getOrElse(az),
+      altitude = altitude.getOrElse(al),
+      zFactor  = z |+| zFactor,
+      target   = target.getOrElse(t)
+    )
+    case e @ FocalSlope(_, z, t) => e.copy(
+      zFactor = z |+| zFactor,
+      target  = target.getOrElse(t)
+    )
+    case e => e
+  }
+}
```

Comment:
This is a complitely user defined logic. Users can define some function that would parse neccesary parameters from the passed arguments, and basing on them it is possible to define the desired MAML Expressions overload. 

The function would recursively walk the MAML tree and would replace nodes that has to be replaced basing on some user defined logic.